test/evp_kdf_test.c

```diff
@@ -245,35 +246,39 @@ static int test_kdf_ss_hash(void)
     };
     unsigned char out[14];
 
-    kctx = EVP_KDF_CTX_new_id(EVP_KDF_SS);
-
+    if ((kctx = EVP_KDF_CTX_new_id(EVP_KDF_SS)) == NULL) {
```

Comment:
TEST_ptr seems more appropriate here (and other similar places below).

Reply:
Hmm I was using the pattern that exists in the whole file. Do you want me to get rid of all the TEST_error() lines?

Reply:
I did spot that this file isn't doing it right. Ideally we should be removing the TEST_error() lines with TEST_int_le, TEST_int_eq etc. So it would be good if you did do that - but I don't require it for approval of this PR.

Reply:
I have removed them